utils/cpuload/conn.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cpuload
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"os"
+	"syscall"
+)
+
+type Connection struct {
+	fd   int
+	pid  uint32
+	seq  uint32
+	addr syscall.SockaddrNetlink
+	rbuf *bufio.Reader
+}
+
+// Create and bind a new netlink socket.
+func NewConnection() (*Connection, error) {
```

Comment:
I didn't do that yet, because we only implemented the two interfaces we needed for getting cpu load. If we need more, I think we can export a more elaborate netlink implementation. There are few in github, but none of those directly expose the task stats stuff.

I moved most of the netlink related stuff in a single file, so we can move/replace it if needed.